project/cmake/scripts/common/macros.cmake

```diff
@@ -62,20 +62,20 @@ endfunction()
 #   (if NO_INSTALL is not given).
 function(copy_file_to_buildtree file relative)
   cmake_parse_arguments(arg "NO_INSTALL" "" "" ${ARGN})
-  if(NOT WIN32)
-    string(REPLACE "\(" "\\(" file ${file})
-    string(REPLACE "\)" "\\)" file ${file})
-  endif()
   string(REPLACE "${relative}/" "" outfile ${file})
+  get_filename_component(outdir ${outfile} DIRECTORY)
 
   if(NOT TARGET export-files)
-    add_custom_target(export-files ALL COMMENT "Copying files into build tree")
+    file(REMOVE ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/ExportFiles.cmake)
+    add_custom_target(export-files ALL COMMENT "Copying files into build tree"
+                      COMMAND ${CMAKE_COMMAND} -P ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/ExportFiles.cmake)
   endif()
   if(NOT ${CORE_SOURCE_DIR} MATCHES ${CMAKE_BINARY_DIR})
     if(VERBOSE)
       message(STATUS "copy_file_to_buildtree - copying file: ${file} -> ${CMAKE_CURRENT_BINARY_DIR}/${outfile}")
     endif()
-    add_custom_command(TARGET export-files COMMAND ${CMAKE_COMMAND} -E copy_if_different "${file}" "${CMAKE_CURRENT_BINARY_DIR}/${outfile}")
+    file(APPEND ${CMAKE_BINARY_DIR}/${CORE_BUILD_DIR}/ExportFiles.cmake
+         "file(COPY \"${file}\" DESTINATION \"${outdir}\")\n")
```

Comment:
file(COPY) always copied back when i wrote this shit (it wasn't supposed to but 2.8.7 had a bug..) my original code did exactly what you do now. i added the copy_if_different thing to workaround the bug. just to defend my sanity ;)